io-engine/src/core/reactor.rs

```diff
@@ -497,19 +498,66 @@ impl Reactor {
         let threads = self.threads.borrow();
         for _ in 0 .. times {
             threads.iter().for_each(|t| {
-                t.poll();
+                self.checked_poll(t);
             });
         }
 
         self.receive_futures();
         self.run_futures();
+
         drop(threads);
 
+        self.add_incoming();
+    }
+
+    /// Polls a thread, checking poll duration.
+    fn checked_poll(&self, t: &spdk_rs::Thread) {
```

Comment:
hmm do we really want to enable this by default? AFAIK this will do a system call? If so this doesn't seem like something we want to do by default for every poll?

Reply:
Ah, I forgot to remove it. Reactor freeze detector essentially does the same, but better.